packages/core/utils/src/product/csv-normalizer.ts

```diff
@@ -0,0 +1,498 @@
+import { isPresent } from "../common"
+import { AdminCreateProduct, AdminCreateProductVariant } from "@medusajs/types"
+
+/**
+ * Column processor is a function that process the CSV column
+ * and writes its value to the output
+ */
+type ColumnProcessor<Output> = (
+  csvRow: Record<string, string | boolean | number>,
+  rowColumns: string[],
+  output: Output
+) => void
+
+/**
+ * Collection of static product columns whose values must be copied
+ * as it is without any further processing.
+ */
+const productStaticColumns: {
```

Comment:
While going through these columns, you might feel why are we repeating all this code that looks almost the same for all the columns.

The goal here is to not write terse code. Instead, the goal is to be descriptive with the fields we handle within the CSV and localize the logic of how a field is handled/normalized. 

To me, this code is easy to reason about vs going through a loop and writing inline conditions there for certain or using transforms like `snakeCase`, `camelCase` to convert the CSV column to DTO property name. 🙂